WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostActivity.java

```diff
@@ -618,48 +617,7 @@ private void initializePostObject() {
             EventBus.getDefault().postSticky(new PostEvents.PostOpenedInEditor(mEditPostRepository.getLocalSiteId(),
                     mEditPostRepository.getId()));
 
-            // run this purge in the background to not delay Editor initialization
-            new Thread(this::purgeMediaToPostAssociationsIfNotInPostAnymore).start();
-        }
-    }
-
-    private void purgeMediaToPostAssociationsIfNotInPostAnymore() {
-        boolean useAztec = AppPrefs.isAztecEditorEnabled();
-        boolean useGutenberg = AppPrefs.isGutenbergEditorEnabled();
-
-        ArrayList<MediaModel> allMedia = new ArrayList<>();
-        allMedia.addAll(mUploadStore.getFailedMediaForPost(mEditPostRepository.getPost()));
-        allMedia.addAll(mUploadStore.getCompletedMediaForPost(mEditPostRepository.getPost()));
-        allMedia.addAll(mUploadStore.getUploadingMediaForPost(mEditPostRepository.getPost()));
-
-        if (!allMedia.isEmpty()) {
-            HashSet<MediaModel> mediaToDeleteAssociationFor = new HashSet<>();
-            for (MediaModel media : allMedia) {
-                if (useAztec) {
```

Comment:
@hypest I'm a bit confused by this code and I was wondering if you could bring some light into it. 
Both `AppPrefs.isAztecEditorEnabled();` and `AppPrefs.isGutenbergEditorEnabled();` always return `true`. Which means we always perform the search for media using `AztecEditorFragment.isMediaInPostBody(` and we never use `PostUtils.isMediaInGutenbergPostBody`. Do you think it's desired behaviour and it's just a preparation for the time when we remove Aztec from the app or it's a bug?

Reply:
It might be a combination of "prepare for removing Aztec" and a bug.

It makes sense to have the different conditionals (in preparation of removing Aztec) but the Gutenberg check should probably come first, I think. I'm not super sure though so, let's also ping @mzorz who wrote this to confirm.

Reply:
Looks like a "prepare for removing Aztec" glitch, which then introduced other bugs.

> let's also ping @mzorz who wrote this to confirm.

Not exactly :D looks like it was actually you @hypest  😄, but I certainly appreciate being considered as someone who participated in building this 🙇 (jokes aside I was definitely involved in these conversations about opt-in and such so, thanks for the ping)


I had a look back in code's history, to make sense about existing code and past decisions (git blame is my friend). A bit of searching shows where this comes from:
- This is where both `isGutenbergEditorEnabled()` and `isAztecEditorEnabled()` are hardcoded to return `true`
https://github.com/wordpress-mobile/WordPress-Android/commit/630f454051c69d91e3653f63705bf2c74ddc37d1
- That commit belongs to this PR https://github.com/wordpress-mobile/WordPress-Android/pull/9208

That explains why / what was done, but of course history doesn't say much about future plans, so deferring back to @hypest here looks like the right way to go.

**What to do**
IMO, regarding that piece of code itself (code in `purgeMediaToPostAssociationsIfNotInPostAnymore()`), we should make sure both `AztecEditorFragment.isMediaInPostBody()` runs when in a non-Gutenberg Post, and also make sure that `PostUtils.isMediaInGutenbergPostBody()` gets run for Gutenberg-block-containing Posts, which apparently is not the case anymore since that commit up there.

> It makes sense to have the different conditionals (in preparation of removing Aztec) but the Gutenberg check should probably come first, I think.

In principle, this sounds like  a good approach (inverting the `if` checks to first check for gutenberg, and `else if` check Aztec). Needless to say, all of these cases should be hunted down and scrutinized thoroughly though, to make sure code does what we want it t do.



We should also care checking for other places where these checks are made, for example [here](https://github.com/wordpress-mobile/WordPress-Android/blob/c02c3168ec394edfb758f60dba19217da3e39dac/WordPress/src/main/java/org/wordpress/android/ui/uploads/MediaUploadReadyProcessor.java) 

I'd also suggest doing a search for both keywords as these throw various places where conditions shown won't be met and are worth taking a deep look into (and fix them), could be in a separate PR but most probably still within the context of this migration to UseCase (not sure what the plans are but I'd tend to think this is important)

- [Search for `isAztecEditorEnabled`](https://github.com/wordpress-mobile/WordPress-Android/search?q=isAztecEditorEnabled&unscoped_q=isAztecEditorEnabled)

- [Search for `isGutenbergEditorEnabled`](https://github.com/wordpress-mobile/WordPress-Android/search?q=isGutenbergEditorEnabled&unscoped_q=isGutenbergEditorEnabled)

**Also note**
Another thing worth noting in the context of evaluating this method and its purpose: the actual purge code there (method `purgeMediaToPostAssociationsIfNotInPostAnymore()`) was introduced in https://github.com/wordpress-mobile/WordPress-Android/commit/fe898920f61c728839b9c30cb36f50b5900eb916, belonging to this PR  https://github.com/wordpress-mobile/WordPress-Android/pull/7081, it was needed in case we had lost `MediaUploadModel` instances in FluxC that are no longer valid (for example when an un unexpected close happened during an upload for an image in a Post), leaving things in an inconsistent state (you can check the [usages for `ClearMediaPayload` here](https://github.com/wordpress-mobile/WordPress-Android/search?q=ClearMediaPayload&unscoped_q=ClearMediaPayload))

Hope that helps clarify things a bit!

Reply:
❤️ ❤️ ❤️ Thanks for such a detailed elaboration @mzorz !!

>we should make sure both AztecEditorFragment.isMediaInPostBody() runs when in a non-Gutenberg Post, and also make sure that PostUtils.isMediaInGutenbergPostBody() gets run for Gutenberg-block-containing Posts

Do you think we should check if the post itself contains a gutenberg block using (https://github.com/wordpress-mobile/WordPress-Android/blob/a29f28c7a259594296ab8cd0d9e46327f97584ee/WordPress/src/main/java/org/wordpress/android/ui/posts/PostUtils.java#L391) => so basically replace the if with `if(contentContainsGutenbergBlocks ) PostUtils.isMediaInGutenbergPostBody() else AztecEditorFragment.isMediaInPostBody()` and not use isAztecEditorEnabled/isGutenbergEditorEnabled at all?

>We should also care checking for other places where these checks are made, for example here

I'll do my best checking those as well, but I'm not sure I have enough context to make the right decision.

Reply:
> Thanks for such a detailed elaboration

yw! ❤️ 

> so basically replace the if with if(contentContainsGutenbergBlocks ) PostUtils.isMediaInGutenbergPostBody() else AztecEditorFragment.isMediaInPostBody() and not use isAztecEditorEnabled/isGutenbergEditorEnabled at all?

That sounds good to me! 👍 

> > We should also care checking for other places where these checks are made, for example here

> I'll do my best checking those as well, but I'm not sure I have enough context to make the right decision.

Understood @malinajirka, your willingness is appreciated, maybe this belongs more in the realm of Gutenberg  team? I can help checking your code with @hypest (or someone @hypest may like to point to) if you need 👍

Reply:
I've created a separate issue for the audit - https://github.com/wordpress-mobile/WordPress-Android/issues/10881. I'll look into it this or next week.